src/Functional/Flip.php

```diff
@@ -0,0 +1,45 @@
+<?php
+/**
+ * Copyright (C) 2011-2016 by Ezinwa Okpoechi <brainmaestro@outlook.com>
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+namespace Functional;
+
+/**
+ * Return a version of the given function where the first arguments two arguments are swapped.
+ *
+ * Any other arguments that are supplied are provided in their correct order unmodified.
+ * If one argument is provided, it is passed to the function without change.
+ *
+ * @param callable $function the function you want to flip
+ * @return callable a flipped version of the given function
+ */
+function flip(callable $callback)

```

Comment:
Okay. That makes sense. I could make this more generic by allowing the user to provide the number of arguments to flip. `flip($numArguments, $callback)`. That should handle the issue you mentioned.